testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/KotlinPluginSmokeTest.groovy

```diff
@@ -161,11 +151,10 @@ class KotlinPluginSmokeTest extends AbstractKotlinPluginSmokeTest {
 
         then:
         result.task(':compileJava').outcome == SUCCESS
-        if (VersionNumber.parse(kotlinVersion).baseVersion < VersionNumber.parse("1.9.25")) {
-            result.tasks.collect { it.path } == [':compileGroovy', ':compileKotlin', ':compileJava']
-        } else {
-            result.tasks.collect { it.path } == [':checkKotlinGradlePluginConfigurationErrors', ':compileGroovy', ':compileKotlin', ':compileJava']
-        }
+
+        def tasks = result.tasks.collect { it.path }
+        tasks.contains(":checkKotlinGradlePluginConfigurationErrors")
+        tasks.findAll { it != ":checkKotlinGradlePluginConfigurationErrors" } == [':compileGroovy', ':compileKotlin', ':compileJava']
```

Comment:
🤔 Won't we always be > 1.9.25 now, so why not drop the filtering and just include the check task in the expected list?

Reply:
Yeah this one was a little weird. Its in a separate commit since I thought it deserved its own commit message:
https://github.com/gradle/gradle/pull/32748/commits/72f62eec4a35df3d84514ea418b82de19f4bd69a

Since this PR updates the tested Kotlin versions, it also exposes some flakiness in the task execution order. 

In some cases, `tasks` is:
`[':checkKotlinGradlePluginConfigurationErrors', ':compileGroovy', ':compileKotlin', ':compileJava']`

And some cases is:
``[':compileGroovy', ':checkKotlinGradlePluginConfigurationErrors', ':compileKotlin', ':compileJava']`

This test verifies that `checkKotlinGradlePluginConfigurationErrors` is present as an executed test, and then also verifies that the _remaining_ tasks execute in a consistent order